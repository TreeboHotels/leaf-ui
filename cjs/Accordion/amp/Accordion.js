'use strict';

exports.__esModule = true;

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _AccordionSection = require('./AccordionSection');

var _AccordionSection2 = _interopRequireDefault(_AccordionSection);

var _AccordionSectionTrigger = require('./AccordionSectionTrigger');

var _AccordionSectionTrigger2 = _interopRequireDefault(_AccordionSectionTrigger);

var _AccordionSectionContent = require('./AccordionSectionContent');

var _AccordionSectionContent2 = _interopRequireDefault(_AccordionSectionContent);

var _reactChildrenGroupByType = require('../../utils/reactChildrenGroupByType');

var _reactChildrenGroupByType2 = _interopRequireDefault(_reactChildrenGroupByType);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var groupIndex = 0;

var Accordion = function (_Component) {
  (0, _inherits3.default)(Accordion, _Component);

  function Accordion(props) {
    (0, _classCallCheck3.default)(this, Accordion);

    var _this = (0, _possibleConstructorReturn3.default)(this, _Component.call(this, props));

    _this.getGroupName = function () {
      groupIndex += 1;
      return '__LEAF_UI__accordion-group-' + groupIndex;
    };

    _this.groupName = _this.getGroupName();
    return _this;
  }

  Accordion.prototype.render = function render() {
    var _this2 = this;

    var children = this.props.children;

    var accordionChildrenGroupByType = (0, _reactChildrenGroupByType2.default)(children);
    var noOtherSiblingSections = accordionChildrenGroupByType[_AccordionSection2.default] && accordionChildrenGroupByType[_AccordionSection2.default].length === 1;

    return _react2.default.createElement(
      'div',
      null,
      _react2.default.Children.map(children, function (child) {
        return child && child.type === _AccordionSection2.default ? _react2.default.cloneElement(child, {
          groupName: _this2.groupName,
          noOtherSiblingSections: noOtherSiblingSections
        }) : child;
      })
    );
  };

  return Accordion;
}(_react.Component);

Accordion.propTypes = {
  children: _propTypes2.default.node
};

Accordion.Section = _AccordionSection2.default;
Accordion.Section.Trigger = _AccordionSectionTrigger2.default;
Accordion.Section.Content = _AccordionSectionContent2.default;

exports.default = Accordion;