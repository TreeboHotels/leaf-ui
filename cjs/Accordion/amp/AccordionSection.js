'use strict';

exports.__esModule = true;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _AccordionSectionTrigger = require('./AccordionSectionTrigger');

var _AccordionSectionTrigger2 = _interopRequireDefault(_AccordionSectionTrigger);

var _AccordionSectionContent = require('./AccordionSectionContent');

var _AccordionSectionContent2 = _interopRequireDefault(_AccordionSectionContent);

var _amp = require('../../Icon/amp');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var inputIndex = 0;

var AccordionSectionContainer = (0, _styledComponents2.default)(function (_ref) {
  var inputId = _ref.inputId,
      props = (0, _objectWithoutProperties3.default)(_ref, ['inputId']);
  return _react2.default.createElement('div', props);
}).withConfig({
  displayName: 'AccordionSection__AccordionSectionContainer'
})(['#', '{display:none;&:checked{~ ', '{height:auto;}+ ', '{', '{transform:rotate(180deg);}}}}'], function (props) {
  return props.inputId;
}, /* sc-selector */_AccordionSectionContent2.default, /* sc-selector */_AccordionSectionTrigger2.default, /* sc-selector */_amp.I);

var AccordionSection = function (_Component) {
  (0, _inherits3.default)(AccordionSection, _Component);

  function AccordionSection(props) {
    (0, _classCallCheck3.default)(this, AccordionSection);

    var _this = (0, _possibleConstructorReturn3.default)(this, _Component.call(this, props));

    _this.getInputId = function () {
      inputIndex += 1;
      return '__LEAF_UI__accordion-trigger-input-' + inputIndex;
    };

    _this.getInputType = function (noOtherSiblingSections) {
      return noOtherSiblingSections ? 'checkbox' : 'radio';
    };

    _this.inputId = _this.getInputId();
    return _this;
  }

  AccordionSection.prototype.render = function render() {
    var _this2 = this;

    var _props = this.props,
        groupName = _props.groupName,
        noOtherSiblingSections = _props.noOtherSiblingSections,
        children = _props.children,
        props = (0, _objectWithoutProperties3.default)(_props, ['groupName', 'noOtherSiblingSections', 'children']);


    return _react2.default.createElement(
      AccordionSectionContainer,
      (0, _extends3.default)({
        inputId: this.inputId
      }, props),
      _react2.default.createElement('input', {
        type: this.getInputType(noOtherSiblingSections),
        name: groupName,
        id: this.inputId
      }),
      _react2.default.Children.map(children, function (child) {
        return child && child.type === _AccordionSectionTrigger2.default ? _react2.default.cloneElement(child, {
          htmlFor: _this2.inputId
        }) : child;
      })
    );
  };

  return AccordionSection;
}(_react.Component);

AccordionSection.propTypes = {
  children: _propTypes2.default.node,
  groupName: _propTypes2.default.string,
  noOtherSiblingSections: _propTypes2.default.bool
};

exports.default = AccordionSection;