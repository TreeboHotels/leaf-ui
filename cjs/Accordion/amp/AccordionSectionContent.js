'use strict';

exports.__esModule = true;

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var AccordionSectionContent = (0, _styledComponents2.default)(function (_ref) {
  var children = _ref.children,
      props = (0, _objectWithoutProperties3.default)(_ref, ['children']);
  return _react2.default.createElement(
    'div',
    props,
    children
  );
}).withConfig({
  displayName: 'AccordionSectionContent'
})(['height:0;overflow:hidden;']);

AccordionSectionContent.propTypes = {
  children: _propTypes2.default.node
};

exports.default = AccordionSectionContent;