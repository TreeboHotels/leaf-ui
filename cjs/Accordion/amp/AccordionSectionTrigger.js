'use strict';

exports.__esModule = true;

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var AccordionSectionTrigger = (0, _styledComponents2.default)(function (_ref) {
  var htmlFor = _ref.htmlFor,
      children = _ref.children,
      props = (0, _objectWithoutProperties3.default)(_ref, ['htmlFor', 'children']);
  return _react2.default.createElement(
    'div',
    props,
    _react2.default.createElement(
      'label',
      { htmlFor: htmlFor },
      children
    )
  );
}).withConfig({
  displayName: 'AccordionSectionTrigger'
})(['', ''], '');

AccordionSectionTrigger.propTypes = {
  htmlFor: _propTypes2.default.string,
  children: _propTypes2.default.node
};

exports.default = AccordionSectionTrigger;