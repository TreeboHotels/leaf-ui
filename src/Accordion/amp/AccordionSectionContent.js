import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const AccordionSectionContentContainer = styled.div`
  display: none;
`;

const AccordionSectionContent = (props) => {
  const {
    children,
  } = props;

  return (
    <AccordionSectionContentContainer>
      {children}
    </AccordionSectionContentContainer>
  );
};

AccordionSectionContent.propTypes = {
  children: PropTypes.node,
};

export default AccordionSectionContent;
