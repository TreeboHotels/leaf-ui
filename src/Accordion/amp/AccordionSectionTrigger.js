import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const AccordionSectionTrigger = styled(
  ({
    htmlFor,
    children,
    ...props
  }) => (
    <div {...props}>
      <label htmlFor={htmlFor}>
        {children}
      </label>
    </div>
  ),
)`
  ${''}
`;

AccordionSectionTrigger.propTypes = {
  children: PropTypes.node,
  htmlFor: PropTypes.string,
};

export default AccordionSectionTrigger;
