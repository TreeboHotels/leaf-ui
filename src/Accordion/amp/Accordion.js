import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AccordionSection from './AccordionSection';
import AccordionSectionTrigger from './AccordionSectionTrigger';
import AccordionSectionContent from './AccordionSectionContent';

let index = 0;
const getAccordionGroup = () => {
  index += 1;
  return `__LEAF_UI__accordion-group-${index}`;
};

class Accordion extends Component {
  constructor(props) {
    super(props);
    this.accordionGroup = getAccordionGroup();
  }

  render() {
    const { children } = this.props;
    return (
      <div>
        {
          React.Children.map(children, (child) => child.type.name === 'AccordionSection' ? (
            React.cloneElement(child, {
              accordionGroup: this.accordionGroup,
            })
          ) : child)
        }
      </div>
    );
  }
}

Accordion.propTypes = {
  children: PropTypes.node,
};

Accordion.Section = AccordionSection;
Accordion.Trigger = AccordionSectionTrigger;
Accordion.Content = AccordionSectionContent;

export default Accordion;
