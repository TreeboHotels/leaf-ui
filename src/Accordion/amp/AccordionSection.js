import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { AccordionSectionTriggerContainer } from './AccordionSectionTrigger';
import { AccordionSectionContentContainer } from './AccordionSectionContent';
import { I } from '../../Icon/amp';

let inputIndex = 0;
const getInputId = () => {
  inputIndex += 1;
  return `__LEAF_UI__accordion-trigger-input-${inputIndex}`;
};

const AccordionSectionContainer = styled(
  ({
    inputId,
    ...props
  }) => <div {...props} />,
)`
  > ${/* sc-selector */AccordionSectionContentContainer} {
    display: none;
  }

  #${(props) => props.inputId} {
    display: none;

    &:checked {
      ~ ${/* sc-selector */AccordionSectionContentContainer} {
        display: block;
      }

      + ${/* sc-selector */AccordionSectionTriggerContainer} {
          ${/* sc-selector */I} {
          transform: rotate(180deg);
        }
      }
    }
  }
`;

class AccordionSection extends Component {
  constructor(props) {
    super(props);
    this.inputId = getInputId();
  }

  render() {
    const {
      groupName,
      children,
      ...props
    } = this.props;

    return (
      <AccordionSectionContainer
        inputId={this.inputId}
        {...props}
      >
        <input type="radio" name={groupName} id={this.inputId} />
        {
          React.Children.map(children, (child) => child && child.type.name === 'AccordionSectionTrigger' ? (
            React.cloneElement(child, {
              htmlFor: this.inputId,
            })
          ) : child)
        }
      </AccordionSectionContainer>
    );
  }
}

AccordionSection.propTypes = {
  groupName: PropTypes.string,
  children: PropTypes.node,
};

export default AccordionSection;
