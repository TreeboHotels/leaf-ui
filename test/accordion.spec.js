/**
 * @fileOverview Test accordion sections with snapshots & shallow rendering
 * @author Ram Pasala<ram.pasala7@gmail.com>
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Accordion from '../src/Accordion/amp/Accordion';
import AccordionSection from '../src/Accordion/amp/AccordionSection';
import AccordionSectionContent from '../src/Accordion/amp/AccordionSectionContent';
import AccordionSectionTrigger from '../src/Accordion/amp/AccordionSectionTrigger';

// test accordion sections shallow rendering & snapshots

it('should render Accordion correctly', () => {
  const accordion = shallow(<Accordion />);
  expect(accordion.find('div')).toHaveLength(1);
  expect(accordion).toMatchSnapshot();
});

it('should render AccordionSection correctly', () => {
  const accordionSectionTriggerTree = renderer
    .create(<AccordionSection />)
    .toJSON();
  const accordionSection = shallow(<AccordionSection />);
  expect(accordionSection.find('AccordionSection__AccordionSectionContainer')).toHaveLength(1);
  expect(accordionSection.find('input[id^="__LEAF_UI__accordion-trigger-input"]')).toHaveLength(1);
  expect(accordionSection.find('input[type="radio"]')).toHaveLength(1);
  expect(accordionSectionTriggerTree).toMatchSnapshot();
});

it('should render AccordionSectionContent correctly', () => {
  const accordionSectionTriggerTree = renderer
    .create(<AccordionSectionContent />)
    .toJSON();
  expect(accordionSectionTriggerTree).toMatchSnapshot();
});

it('should render AccordionSectionTrigger correctly', () => {
  const accordionSectionTriggerTree = renderer
    .create(<AccordionSectionTrigger />)
    .toJSON();
  expect(accordionSectionTriggerTree).toMatchSnapshot();
});
