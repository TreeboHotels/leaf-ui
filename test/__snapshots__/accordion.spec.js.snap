// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Accordion correctly 1`] = `<div />`;

exports[`should render AccordionSection correctly 1`] = `
.c0 #__LEAF_UI__accordion-trigger-input-1 {
  display: none;
}

.c0 #__LEAF_UI__accordion-trigger-input-1:checked ~ .AccordionSectionContent-jsuqGi {
  height: auto;
}

.c0 #__LEAF_UI__accordion-trigger-input-1:checked + .AccordionSectionTrigger-caJlXp .Icon__I-gGplGf {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

<div
  className="c0"
>
  <input
    id="__LEAF_UI__accordion-trigger-input-1"
    name={undefined}
    type="radio"
  />
</div>
`;

exports[`should render AccordionSectionContent correctly 1`] = `
.c1 {
  height: 0;
  overflow: hidden;
}

.kJDsUm #__LEAF_UI__accordion-trigger-input-1:checked ~ .c0 {
  height: auto;
}

<div
  className="c0 c1"
/>
`;

exports[`should render AccordionSectionTrigger correctly 1`] = `
.kJDsUm #__LEAF_UI__accordion-trigger-input-1:checked + .c0 .Icon__I-gGplGf {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

<div
  className="c0 "
>
  <label
    htmlFor={undefined}
  />
</div>
`;
